### ISSUE TITLE
tf.nn.silu returns zero second derivative for negative float64 input where analytic result is finite

### ISSUE THREAD

**Issue by ALinrunrun** [tensorflow/tensorflow#124832]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.21.0

### Custom code

Yes

### OS platform and distribution

Linux Ubuntu 22.04

### Python version

Python 3.11

### Current behavior?

`tf.nn.silu` returns an incorrect second derivative for a negative float64 input.

For SiLU / swish, the function is `x * sigmoid(x)`. At `x = -709.0`, the second derivative is very small but still a finite normal float64 value. A 50-digit analytic mpmath reference gives approximately `-8.6026399069076e-306`.

However, TensorFlow autodiff returns `0.0` for this second derivative. This is a categorical error because the expected result is finite and nonzero, with the opposite sign from zero. Peer autodiff backends such as Paddle and PyTorch compute the expected finite value, so this appears to be a higher-order backward-pass issue rather than a float64 limitation.

Expected behavior?

The second derivative of `tf.nn.silu` at `x = -709.0` should be close to the analytic reference value `-8.6026399069076e-306`, not `0.0`.

### Standalone code to reproduce the issue

```shell
import os
os.environ.setdefault("CUDA_VISIBLE_DEVICES", "-1")
os.environ.setdefault("TF_CPP_MIN_LOG_LEVEL", "3")

import tensorflow as tf

x = -709.0
truth = -8.6026399069076e-306
rtol = 1e-6

v = tf.Variable(x, dtype=tf.float64)

with tf.GradientTape() as tape2:
    with tf.GradientTape() as tape1:
        y = tf.nn.silu(v)
    g1 = tape1.gradient(y, v)
g2 = tape2.gradient(g1, v)

actual = float(g2)
rel_err = abs(actual - truth) / abs(truth)

print(f"x: {x!r}")
print(f"first derivative: {float(g1)!r}")
print(f"second derivative: {actual!r}")
print(f"expected second derivative: {truth!r}")
print(f"relative error: {rel_err:.3e}")

if rel_err > rtol:
    print("BUG REPRODUCED: tf.nn.silu second derivative is wrong for negative float64 input")
else:
    print("not reproduced")
```

### Relevant log output

```shell
tf: d^2 silu(-709.0) = 0.0  truth=-8.6026399069076e-306  rel_err=1.000e+00  WRONG
```
